src/System Application/App/AppSource Gallery/src/AppSourceProduct.Table.al

```diff
@@ -0,0 +1,113 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.Apps.AppSource;
+
+/// <summary>
+/// Represents an except of a product definition in AppSource.
+/// </summary>
+table 2515 "AppSource Product"
+{
+    DataClassification = SystemMetadata;
+    Access = Internal;
+    TableType = Temporary;
+
+    InherentEntitlements = RIMDX;
+    InherentPermissions = RIMDX;
+
+    fields
+    {
+        field(1; UniqueProductID; Text[200])
+        {
+            Caption = 'Unique Product ID';
+        }
+        field(2; DisplayName; Text[250])
+        {
+            Caption = 'Name';
+        }
+        field(4; PublisherID; Text[200])
+        {
+            Caption = 'Publisher ID';
+        }
+        field(5; PublisherDisplayName; Text[250])
+        {
+            Caption = 'Publisher Name';
+        }
+        field(6; PublisherType; Text[200])
+        {
+            Caption = 'Publisher Type';
+        }
+        field(8; RatingAverage; Decimal)
+        {
+            Caption = 'Average Rating';
+        }
+        field(9; RatingCount; Integer)
+        {
+            Caption = '# Ratings';
+        }
+        field(10; ProductType; Text[200])
+        {
+            Caption = 'Product Type';
+        }
+        field(11; AppID; Text[36])
+        {
+            Caption = 'Application Identifier';
+        }
+        field(12; Popularity; Decimal)
+        {
+            Caption = 'Popularity';
+        }
+        field(14; LastModifiedDateTime; DateTime)
+        {
+            Caption = 'Last Modified Date Time';
+        }
+    }
+
+    keys
+    {
+        key(UniqueID; UniqueProductID)
+        {
+
+        }
+        key(DefaultSorting; DisplayName, PublisherDisplayName)
+        {
+
+        }
+        key(Rating; RatingAverage, DisplayName, PublisherDisplayName)
+        {
+
+        }
+        key(Popularity; Popularity, DisplayName, PublisherDisplayName)
+        {
+
+        }
+        key(LastModified; LastModifiedDateTime, DisplayName, PublisherDisplayName)
+        {
+
+        }
+        key(PublisherID; PublisherID, DisplayName, PublisherDisplayName)
+        {
+
+        }
+    }
+
+    fieldgroups
+    {
+        fieldgroup(Brick; PublisherDisplayName, DisplayName, Popularity, RatingAverage)
+        {
+        }
+    }
+
+    procedure ReloadAllProducts()
+    var
+        AppSourceProductManager: codeunit "AppSource Product Manager";
+    begin
+        Clear(Rec);
```

Comment:
I don't think we should remove all filters on the rec.. I would instead create another temp record, then share table with Rec and update that one. This way any filters, sorting etc. the user has applied are maintained. We should not override user input.